File: Connect Nodes at Same Level.cpp
class Solution{
    public:
    void connect(Node *root)
    {
       // Code Here
       queue<Node *> q;
       
       q.push(root);
       
       while(!q.empty())
       {
           int n = q.size();
           vector<Node* > temp;
           
           Node* conn = NULL;
           
           for(int i = 0; i < n ; i++)
           {
               
               auto f = q.front();
               q.pop();
               
               
               if(f->left) q.push(f->left);
               if(f->right) q.push(f->right);
               
               if(i == 0)
               {
                   conn = f;
               }
               
               else
               {
                   conn->nextRight = f;
                   conn = f;
               }
           }
           
         
       }
    }    
};

File: Connect Nodes at Same Level.java
class Solution {

    public void connect(Node root) {

        

        Queue<Node> q = new LinkedList<>();

        q.offer(root);

        while(!q.isEmpty()) {

            int levelSize = q.size();

            Node prev = null;

            for(int i = 0; i < levelSize; i++) {

                Node curr = q.poll();

                if(curr.left != null) {

                    q.offer(curr.left);

                }

                if(curr.right != null) {

                    q.offer(curr.right);

                }

                if(prev != null) {

                    prev.nextRight = curr;

                }

                prev = curr;

            }

        }

    }

}
